### PR TITLE
DOC Update doc structure for docs.silverstripe.org

### PR DESCRIPTION
**Pull request by GuySartorelli**
Also remove unnecessary docs and add doc linting

## Issue 
- https://github.com/silverstripe/silverstripe-taxonomy/issues/107

### CODE REVIEW

File: docs/en/01_basic_usage.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    // ...
+    private static $many_many = [
+        'Terms' => TaxonomyTerm::class
+    ];
+
+    public function getCMSFields()
+    {
+        $fields = parent::getCMSFields();
+        // ...
+        $fields->addFieldToTab('Root.Main', TreeMultiselectField::create(
```
Comment (ChloeSartorelli) [inline docs/en/01_basic_usage.md:33]: `TreeMultiselectField` is way more intuitive and lightweight than the old docs saying to use `GridField`.

**Review by GuySartorelli — COMMENTED** [review]
Removed the following documentation:

|section|reason|
|---|---|
|Filtering with Taxonomy Terms|Shows a pretty complex query for a very narrow use case that avoids using the ORM (so, narrow use case with arguably bad practices)|
|Taxonomy Navigation|Seems unlikely you'd want a whole page dedicated to this - you'd probably more likely just want to add this as a part of a pre-existing page.<br>Even if you do want this, it's pretty straight forward to create a page.<br>We'll deprecate the built in `TaxonomyDirectoryController` - that only works for the `Page` class and only when a relation with a specific name is set up, it's too narrow in purpose|

File: composer.json [hunk 2]
```diff
@@ -35,7 +36,11 @@
             "SilverStripe\\Taxonomy\\Tests\\Behat\\Context\\": "tests/behat/src/"
         }
     },
-    "extra": [],
```
Comment (ChloeSartorelli) [inline composer.json:38]: Removed unnecessary empty section

File: docs/en/userguide/index.md
Comment (ChloeSartorelli) [inline docs/en/userguide/index.md:1]: All changes in this file were just to pass linting.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**